File: service/src/main/java/com/db/service/state/SuspendedStudentState.java
package com.db.service.state;

import org.springframework.stereotype.Component;

@Component
public class SuspendedStudentState implements StudentState {
    @Override
    public boolean canRegisterCourse() {
        return false;
    }

    @Override
    public String getStatusName() {
        return "SUSPENDED";
    }
}

File: ui/src/app/core/state/impl/suspended-student.state.ts
import { Injectable } from '@angular/core';
import { StudentState } from '../student-state';

@Injectable({ providedIn: 'root' })
export class SuspendedStudentState implements StudentState {
  getStatusName(): string { return 'SUSPENDED'; }
  canEdit(): boolean { return true; }
  canRegisterCourse(): boolean { return false; }
  canRestore(): boolean { return false; }
}
